component/grpc/observability.go

```diff
@@ -102,6 +103,16 @@ func (o *observer) log(err error) {
 	}
 	if err != nil {
 		fields["error"] = err.Error()
+
+		if ef, ok := err.(errors.WithFields); ok {
```

Comment:
a _proper_ solution would unwrap errors all the way down to get fields from wrapped errors at every level.